open-hackathon-server/src/hackathon/storage/azure_storage.py

```diff
@@ -22,13 +22,136 @@
+        blob_name = self.__generate_file_name(context.hackathon_name, context.file_name)
+
+        if context.get('content'):
+            file_content = context.content
+            result = self.file_service.upload_file_to_azure(file_content, container_name, blob_name)
+        else:
+            assert context.get('file_path')
+            file_path = context.file_path
+            result = self.file_service.upload_file_to_azure_from_path(file_path, container_name, blob_name)
+
+        if result is not None:
+            self.cache_manager.invalidate(result)
+        context["url"] = result
+        self.log.debug("File saved at:" + result)
+        return context
+
+    def load(self, context):
+        """Load file from storage
+
+        :type context: Context
+        :param context: the execution context of file loading
+
+        :rtype: dict
+        :return: the file content
+        """
+        assert context.get("url")
+        print context.url, context.path
+
+        def get_temp():
+            temp_path = self.file_service.download_file_from_azure(azure_file_url=context.url,
+                                                                   local_file_path=context.path)
+            if temp_path is None:
+                return None
+            if context.file_type == FILE_TYPE.TEMPLATE:
+                with open(temp_path) as template_file:
+                    return json.load(template_file)
+            else:
+                return None
+
+        return self.cache_manager.get_cache(key=context.url, createfunc=get_temp)
+
+    def delete(self, context):
+        """Delete file from Azure storage
+
+        :type context: Context
+        :param context: must have {"url":""} key or {"container_name":"***","blob_name":"***"} keys
+
+        :rtype: bool
+        :return: True if successfully deleted, otherwise False
+        """
+        try:
+            if context.get("url"):
+                url_arr = context.url.split('/')
+                file_name = url_arr[-1]
+                hackathon_name = url_arr[-2]
+                container_name = url_arr[-3]
+                blob_name = hackathon_name + "/" + file_name
+                print container_name, blob_name
+            else:
+                assert context.get("container_name") and context.get("blob_name")
+                container_name = context.container_name
+                blob_name = context.blob_name
+            self.file_service.delete_file_from_azure(container_name, blob_name)
+            url = self.file_service.blob_service.make_blob_url(container_name, blob_name)
+            self.cache_manager.invalidate(url)
+            return True
+        except Exception as e:
+            self.log.error(e)
+            return False
+
+    def report_health(self):
+        """The status of Azure storage should be always True"""
+        try:
+            if self.create_container_in_storage('images', 'container'):
```

Comment:
It should be `self.file_service.create_container_in_storage(...)`. Method `create_container_in_storage` is defined in `file_service`.
Please make sure ALL methods are tested